sigstore/_internal/rekor/_client.py

```diff
@@ -2,8 +2,95 @@
 Client implementation for interacting with Rekor.
 """
 
+import json
+from abc import ABC
+from typing import Any, Optional, Tuple
+from urllib.parse import urljoin
+
+import requests  # type: ignore
+
+DEFAULT_REKOR_URL = "https://rekor.sigstore.dev/api/v1/"
+
+
+class Endpoint(ABC):
```

Comment:
I removed the `get` and `post` methods that returned `NotImplementedError`. MyPy didn't seem to like them since the subclasses of `Endpoint` had the methods taking different args.